core/tests/unittests/bayesopt/gpautograd/test_comparison_gpy.py

```diff
@@ -0,0 +1,328 @@
+        COVARIANCE_SCALE_LOWER_BOUND, COVARIANCE_SCALE_UPPER_BOUND,
+        warning=False)
+    # Normalize targets to mean 0, variance 1 (this is done in our code
+    # internally)
+    targets_mean = np.mean(data['train_targets'])
+    targets_std = max(np.std(data['train_targets']), 1e-8)
+    targets_normalized = data['train_targets_normalized']
+    model = GPy.models.GPRegression(
+        data['train_inputs'], targets_normalized.reshape((-1, 1)),
+        kernel=_kernel, noise_var=INITIAL_NOISE_VARIANCE)
+    model.likelihood.variance.constrain_bounded(
+        NOISE_VARIANCE_LOWER_BOUND, NOISE_VARIANCE_UPPER_BOUND,
+        warning=False)
+    # Note: We could also set hyperpriors somehow, using model.priors. But
+    # this is pretty undocumented!
+    # Fit hyperparameters
+    # Note: Should set optimization_config.lbfgs_tol here, but don't know how
+    np.random.seed(random_seed)  # Seed plays very different role here. Whatever...
+    verbose = optimization_config.verbose
+    model.optimize_restarts(
+        num_restarts=optimization_config.n_starts,
+        optimizer='bfgs',
+        max_iters=optimization_config.lbfgs_maxiter,
+        verbose=verbose)
+    # Print hyperparameter values
+    print(model)
+    print('\n' + str(_kernel.lengthscale))
+    # Prediction (have to be rescaled to undo normalization)
+    means, vars = model.predict(
+        data['test_inputs'], include_likelihood=False)
+    means = means * targets_std + targets_mean
+    stddevs = np.sqrt(vars) * targets_std
+    return {
+        'means': means,
+        'stddevs': stddevs}
+
+
+def _plot_comparison(y_list: List[np.ndarray]):
+    yshape = y_list[0].shape
+    assert all(y.shape == yshape for y in y_list)
+    assert len(yshape) == 2, "Can only do 2D plots"
+    min_val = min([np.min(y) for y in y_list])
+    separator = np.ones((yshape[0], 5)) * min_val
+    ys_and_seps = [x for l in zip(y_list, [separator] * len(y_list)) for x in l]
+    ys_and_seps = ys_and_seps[:-1]
+    plt.imshow(np.concatenate(ys_and_seps, axis=1))
+    plt.colorbar()
+
+
+# Note: this function is not ran by the tests but can be executed separately
+# to get a visual interpretation, see `test_comparison_gpy`
+def plot_predictions(
+        data: dict, pred_ours: dict, pred_gpy: dict, title: str):
+    grid_shape = data['grid_shape']
+    plt.title(title)
+    for i, key in enumerate(('means', 'stddevs')):
+        plt.subplot(2, 1, i + 1)
+        lst = [pred_ours[key].reshape(grid_shape),
+               pred_gpy[key].reshape(grid_shape)]
+        if i == 0:
+            lst.insert(1, data['true_targets'].reshape(grid_shape))
+        _plot_comparison(lst)
+        plt.title(key)
+    plt.show()
+
+
+SRC_URL = 'https://autogluon.s3.amazonaws.com'
+TRG_PATH = os.path.join('tests', 'unittests', 'bayesopt', 'gpautograd')
+
+
+def download_pickle_file(fname):
+    trg_fname = os.path.join(TRG_PATH, 'numcomp', fname)
+    if not os.path.exists(trg_fname):
```

Comment:
it may become nasty if the working directory is occupied by another process, I will use tempfile instead for temporary files

Reply:
Is there an example somewhere else in the code? I'll grep for tempfile

Reply:
Done